strings/aho_corasick.py

```diff
@@ -22,8 +29,10 @@ def find_next_state(self, current_state: int, char: str) -> Union[int, None]:
     def add_keyword(self, keyword: str) -> None:
         current_state = 0
         for character in keyword:
-            if self.find_next_state(current_state, character):
-                current_state = self.find_next_state(current_state, character)
+            next_state = self.find_next_state(current_state, character)
+            if next_state:
+                assert next_state is not None
```

Comment:
Can you tell me why this change was necessary? It seems perfectly fine as it was.

If `mypy` was giving any errors then make the below changes:
```suggestion
            if next_state is not None:
```

Reply:
Okay, I have removed all such instances and have restructured the logic as needed

Reply:
Actually I didn't want to change the logic of the statements, that's why I was using `assert`